kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/GCSToBQWriter.java

```diff
@@ -108,23 +109,28 @@ public void writeRows(List<RowToInsert> rows,
     }
 
     int retryCount = 0;
-    boolean exceptionsOccurred;
+    boolean success = false;
     do {
       if (retryCount > 0) {
         waitRandomTime();
       }
-      exceptionsOccurred = false;
-      // Perform GCS Upload and BQ Transfer
+      // Perform GCS Upload
       try {
         uploadRowsToGcs(rows, blobInfo);
-      } catch (ConnectException ce) {
-        exceptionsOccurred = true;
+        success = true;
+      } catch (StorageException se) {
         logger.warn("Exceptions occurred for table {}, attempting retry", tableId.getTable());
       }
       retryCount++;
-    } while (exceptionsOccurred && (retryCount < retries));
+    } while (!success && (retryCount < retries));
+
+    if (success) {
+      logger.info("Batch loaded {} rows", rows.size());
+    }
+    else {
+      throw new ConnectException(String.format("Failed to load %d rows into GCS within %d attempts.", rows.size(), retries));
```

Comment:
Similarly here and in line 128 (logger.info("Batch loaded...).  It would be nice to be able to identify the connector that is generating the error.

Reply:
MDC. See above.